src/project-type/workspace-instance.ts

```diff
@@ -1,18 +1,38 @@
-import { WorkspaceApi } from "@sap/project-api";
-
-// TODO: define subset of properties signatures
-let workspaceAPIProxy: WorkspaceApi;
+import { ProjectApi, Tag, WorkspaceApi } from "@sap/project-api";
+import { Uri, WorkspaceFolder } from "vscode";
+import { BasWorkspaceApi } from "../../types/api";
 
+let basWSAPI: BasWorkspaceApi;
 /**
  * @param WorkspaceImpl - constructor for the `WorkspaceApi`
  *                        dependency injection is used to enable easier testing
  */
-export function initWorkspaceAPI(WorkspaceImpl: { new (): WorkspaceApi }) {
-  workspaceAPIProxy = new WorkspaceImpl();
-  // TODO: implement "READ-ONLY" proxy
+export function getBasWorkspaceAPI(WorkspaceImpl: { new(): WorkspaceApi }): BasWorkspaceApi {
+  if (!basWSAPI) {
+    const workspaceImpl = new WorkspaceImpl();
+
+    initWorkspaceImpl(workspaceImpl);
+
+    basWSAPI = createWorkspaceReadOnlyProxy(workspaceImpl);
+  }
+
+  return basWSAPI;
 }
 
-// TODO: define subset of properties signatures
-export function getWorkspaceAPI(): WorkspaceApi {
-  return workspaceAPIProxy;
+
+function initWorkspaceImpl(workspaceImpl: WorkspaceApi): void {
+  workspaceImpl.startWatch();
 }
+
+function createWorkspaceReadOnlyProxy(workspaceImpl: WorkspaceApi): BasWorkspaceApi {
+  const basWSAPI: BasWorkspaceApi = {
+    getProjects: (tag?: Tag): Promise<ProjectApi[]> => workspaceImpl.getProjects(tag),
```

Comment:
I wonder if there is a way for us to avoid copying the signatures of the `WorkspaceApi`

Reply:
Just to disable eslint error, please check but I can revert to the previous logic